alter StdWXCalculate to add optional loop,archive bindings to directives

g-eddy opened the pull request:
* hopefully i got branch, commits, etc correct this time. i replaced the fork and worked with that
* decision on which calculations to make have been moved to new_loop_packet() and new_archive_record(), by splitting the calculations list into two, worked out at configuration time
* lists instead of comma-separated strings used internally
* i haven't changed Manual as that is part of 4.3.0beta whereas i am thinking in terms of 4.2.0 as base
* i noticed during testing that ET returned a None value which was faithfully inserted into the packet. this is a minor defect outside scope of this change

tkeffer commented:
You're on the right track. I've modified your PR, trying to simplify it, and make it a bit more "WeeWX-like." I've also added documentation to the User's Guide.

We cannot use f-strings, because they were introduced in Python 3.6, and we are targeting 2.7 and later, so I changed back to traditional formatting. 

Take a look and see if the changes are to your liking. Branch https://github.com/weewx/weewx/tree/g-eddy-development

Thanks! This will be a solid addition.

g-eddy commented:
fine with me. i’ll pick up the style and utilities (and that github thingy) over time. cheers
____________
Graham Eddy

> On 7 Jan 2021, at 12:57 am, Tom Keffer <notifications@github.com> wrote:
> 
> 
> You're on the right track. I've modified your PR, trying to simplify it, and make it a bit more "WeeWX-like." I've also added documentation to the User's Guide.
> 
> We cannot use f-strings, because they were introduced in Python 3.6, and we are targeting 2.7 and later, so I changed back to traditional formatting.
> 
> Take a look and see if the changes are to your liking. Branch https://github.com/weewx/weewx/tree/g-eddy-development <https://github.com/weewx/weewx/tree/g-eddy-development>
> Thanks! This will be a solid addition.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/weewx/weewx/pull/630#issuecomment-755313182>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APQPY4XPQTNXBAB54X3CAU3SYRT5XANCNFSM4VL3M3RA>.
>
